### PR TITLE
Python Test Fix and Refactor

### PR DESCRIPTION
**Pull request by drewjj**
- Fixes issues with current develop branch Python tests
- Refactors all Python code to be in a services directory
- Refactor pgquery.py to be in a common directory so it can be used by all images that need to query PostgreSQL
- Rework Dockerfiles to get the new file structure to work
- Update all requirements.txt files and upgrade to latest libraries

### CODE REVIEW

File: services/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+sqlalchemy==2.0.21
+pg8000==1.30.2
+python-dateutil==2.8.2
+flask==3.0.0
+flask_restful==0.3.10
+marshmallow==3.20.1
+gunicorn==21.2.0
+pytz==2023.3.post1
+Werkzeug==3.0.0
+uuid==1.30
```
Comment (mwodahl) [inline services/requirements.txt:23]: I had to also add multidict to this requirements file when running tests

**Review by mwodahl — COMMENTED** [review]
I just had the one comment about the requirements.txt file - otherwise tests look good and all pass!

File: services/requirements.txt (hunk 1 above)
Comment (mwodahl) [reply services/requirements.txt:23]: This is now added

**Comment by drewjj** [comment]
Cool, verified the dependencies are all good now from a fresh Linux box.

**Review by mwodahl — APPROVED** [review]

**Comment by drewjj** [comment]
Okay, this pull request is good to go for review again

**Review by payneBrandon — APPROVED** [review]

**Pull request merged**